File: app/csharp/simple/Services/Helpers/HelperMethods.cs
using System;
using System.Threading.Tasks;

using Azure.Identity;
using Azure.Security.KeyVault.Secrets;

namespace Simple.Services.Helpers
{
    /// <summary>
    /// 様々な操作のためのヘルパーメソッドを提供します。
    /// </summary>
    public static class HelperMethods
    {
        /// <summary>
        /// Azure Key Vaultからシークレットを取得します。
        /// </summary>
        /// <param name="secretName">取得するシークレットの名前。</param>
        /// <returns>シークレットの値。</returns>
        public static string GetSecretFromKeyVault(string secretName)
        {
            return "DUMMY_SECRET";  // ビルドエラーを回避するため、ダミーの値を返す
            // string keyVaultName = Environment.GetEnvironmentVariable("AZURE_KEY_VAULT_NAME");
            // string keyVaultUri = $"https://{keyVaultName}.vault.azure.net";
            // var credential = new DefaultAzureCredential();
            // var client = new SecretClient(new Uri(keyVaultUri), credential);
            // KeyVaultSecret secret = client.GetSecret(secretName);
            // return secret.Value;
        }
    }
}

File: app/python/simple/service/helpers/helper_methods.py
import os
from azure.keyvault.secrets import SecretClient
from azure.identity import DefaultAzureCredential


def get_secret_from_key_vault(secret_name: str) -> str:
    """
    Azure Key Vaultからシークレットを取得します。

    Args:
        secret_name (str): 取得するシークレットの名前。

    Returns:
        str: シークレットの値。
    """
    return "dummy_secret"  # ビルドエラー回避のため、ダミーの値を返す
    # key_vault_name = os.getenv("AZURE_KEY_VAULT_NAME")
    # key_vault_uri = f"https://{key_vault_name}.vault.azure.net"
    # credential = DefaultAzureCredential()
    # client = SecretClient(vault_url=key_vault_uri, credential=credential)
    # secret = client.get_secret(secret_name)
    # return secret.value
